packages/cells/src/model.ts

```diff
@@ -654,6 +656,7 @@ export class CodeCellModel extends CellModel implements ICodeCellModel {
     globalModelDBMutex(() => {
       const sharedCell = this.sharedModel as models.ISharedCodeCell;
       sharedCell.setOutputs(outputs);
+      sharedCell.execution_count = execution_count;
```

Comment:
Same here, we should never initialize `execution_count` in the constructor of some class. Content should be initialized in the `create`/`createStandalone` methods of the shared model. Otherwise, we will always initialize to 0 when a new client joins the session.

I'm a bit puzzled why we initialize the outputs above..